new MyPromise((resolve, reject) => {
  reject("error");
})
  .then((response) => {
    console.log(`Hi, ${response}`);
    return "bye";
  })
  .then(() => {
    console.log("then");
  })
  .catch((error) => {
    console.log(error);
    return "Hello World!"
  })
  .then((response) => {
    console.log(response);
  })
  .catch(console.log);

new MyPromise((resolve, reject) => {
  setTimeout(() => reject("error"), 2000);
})
  .then((response) => {
    console.log(`Hi, ${response}`);
    return "bye";
  })
  .then(() => {
    console.log("then");
  })
  .catch((error) => {
    console.log(error);
    return "Hello World!"
  })
  .then((response) => {
    console.log(response);
  })
  .catch(console.log);